src/praisonai-train/praisonai_train/train/_ollama.py

```
@@ -5,93 +5,221 @@
+
     # Create and push model
     tag = f"{ollama_model}:{model_parameters}"
-    
-    subprocess.run(["ollama", "create", tag, "-f", "Modelfile"], check=True)
-    subprocess.run(["ollama", "push", tag], check=True)
\ No newline at end of file
+
+    create_cmd = ["ollama", "create", tag, "-f", "Modelfile"]
+    if quantization:
+        create_cmd += ["--quantize", quantization]
+    subprocess.run(create_cmd, check=True)
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)